linkerd/protocol/h2/src/main/scala/io/buoyant/linkerd/protocol/h2/grpc/GrpcClassifier.scala

```diff
@@ -0,0 +1,94 @@
+package io.buoyant.linkerd.protocol.h2.grpc
+
+import com.twitter.finagle.buoyant.h2.service.{H2Classifier, H2ReqRep, H2ReqRepFrame}
+import com.twitter.finagle.service.ResponseClass
+import com.twitter.util.Return
+import io.buoyant.grpc.runtime.GrpcStatus
+import io.buoyant.grpc.runtime.GrpcStatus.{Ok, Unavailable}
+
+
+trait GrpcClassifier extends H2Classifier {
+  /**
+    * @inheritdoc
+    * Since GRPC sends status codes in the
+    * [[com.twitter.finagle.buoyant.h2.Frame.Trailers Trailers]] frame of an H2
+    * stream, we can never attempt early classification
+    */
+  override val responseClassifier: PartialFunction[H2ReqRep, ResponseClass] =
+    PartialFunction.empty
```

Comment:
We can classify Throws here as non-retryable failures (there will be no stream).